phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -768,7 +768,17 @@ private boolean avoidRpcToGetTable(boolean alwaysHitServer, Long resolvedTimesta
             // What if the table is created with UPDATE_CACHE_FREQUENCY explicitly set to ALWAYS?
             // i.e. explicitly set to 0. We should ideally be checking for something like
             // hasUpdateCacheFrequency().
-            if (table.getUpdateCacheFrequency() != 0L) {
+            //always fetch an Index in PENDING_DISABLE state to retrieve server timestamp
+            //QueryOptimizer needs that to decide whether the index can be used
+            if (PIndexState.PENDING_DISABLE.equals(table.getIndexState())) {
+                effectiveUpdateCacheFreq =
+                        (Long) ConnectionProperty.UPDATE_CACHE_FREQUENCY.getValue(
+                            connection.getQueryServices().getProps().get(
+                  QueryServices.UPDATE_CACHE_FREQUENCY_FOR_PENDING_DISABLED_INDEX,
+                  QueryServicesOptions.DEFAULT_UPDATE_CACHE_FREQUENCY_FOR_PENDING_DISABLED_INDEX));
+                ucfInfoForLogging = "pending-disable-index-level";
+            } else if (table.getUpdateCacheFrequency()
+                    != QueryServicesOptions.DEFAULT_UPDATE_CACHE_FREQUENCY) {
```

Comment:
Should this be:

```
else if (table.getUpdateCacheFrequency() != 0)
```
>>
      // What if the table is created with UPDATE_CACHE_FREQUENCY explicitly set to ALWAYS?
      // i.e. explicitly set to 0. We should ideally be checking for something like
      // hasUpdateCacheFrequency().
It specifically checks whether table.getUpdateCacheFrequency is set to ALWAYS or not?

Reply:
@shahrs87 Actually they check that the table has UCF set to anything other than the default value - if yes, they take the effective UCF as the table-level property and also set `ucfInfoForLogging = "table-level"`. 

The comment highlights the fact that we can not say for sure whether a table was created with UCF explicitly set to the default value or we just assigned it the default value since there was no value defined at creation time i.e. we do not have a way to tell if a table `hasUpdateCacheFrequency()`.